runtime/compiler/env/j9methodServer.cpp

```diff
@@ -1935,8 +2010,233 @@ TR_ResolvedRelocatableJ9JITaaSServerMethod::getUnresolvedSpecialMethodInCP(I_32
+   J9ConstantPool *constantPool = (J9ConstantPool *)(J9_CP_FROM_METHOD(ramMethod()));
+
+   bool fieldInfoCanBeUsed = false;
+
+   _stream->write(JITaaS::J9ServerMessageType::ResolvedRelocatableMethod_ResolveStaticFieldRefWithMethod, ramMethod(), getRemoteMirror(), cpIndex, isStore);
+   auto recv = _stream->read<void *, bool, UDATA >();
+   offset = std::get<0>(recv);
+   *unresolvedInCP = std::get<1>(recv);
+   ltype = std::get<2>(recv);
+
+   if (needAOTValidation)
+      {
+      if (comp->getOption(TR_UseSymbolValidationManager))
+         {
+         TR_J9VMBase *fej9 = (TR_J9VMBase *) comp->fe();
+         TR::CompilationInfo *compInfo = TR::CompilationInfo::get(fej9->_jitConfig);
+         TR_RelocationRuntime *reloRuntime = compInfo->getCompInfoForThread(fej9->vmThread())->reloRuntime();
+
+         TR_OpaqueClassBlock *clazz = reloRuntime->getClassFromCP(fej9->vmThread(), fej9->_jitConfig->javaVM, constantPool, cpIndex, true);
+
+         fieldInfoCanBeUsed = comp->getSymbolValidationManager()->addDefiningClassFromCPRecord(clazz, constantPool, cpIndex, true);
+         }
+      else
+         {
+         fieldInfoCanBeUsed = storeValidationRecordIfNecessary(comp, constantPool, cpIndex, TR_ValidateStaticField, ramMethod());
+         }
+      }
+   else
+      {
+      fieldInfoCanBeUsed = true;
+      }
+
+   if (offset == (void *)J9JIT_RESOLVE_FAIL_COMPILE)
+      {
+      comp->failCompilation<TR::CompilationException>("offset == J9JIT_RESOLVE_FAIL_COMPILE");
+      }
+
+   if (offset && fieldInfoCanBeUsed &&
+      (!(_fe->_jitConfig->runtimeFlags & J9JIT_RUNTIME_RESOLVE) ||
+      comp->ilGenRequest().details().isMethodHandleThunk() || // cmvc 195373
+      !performTransformation(comp, "Setting as unresolved static attributes cpIndex=%d\n",cpIndex)) /*&&
+      compInfo->isRomClassForMethodInSharedCache(method, jitConfig->javaVM) */  /* &&
+      fieldIsFromLocalClass((void *)_methodCookie, cpIndex)*/)
+      {
+      theFieldIsFromLocalClass = true;
+      ltype = fieldShape->modifiers;
+      volatileFlag = (ltype & J9AccVolatile) ? 1 : 0;
+      finalFlag = (ltype & J9AccFinal) ? 1 : 0;
+      privateFlag = (ltype & J9AccPrivate) ? 1 : 0;
+      *address = offset;
+
+      }
+   else
+      {
+#if defined(J9VM_INTERP_AOT_COMPILE_SUPPORT) && defined(J9VM_OPT_SHARED_CLASSES) && (defined(TR_HOST_X86) || defined(TR_HOST_POWER) || defined(TR_HOST_S390) || defined(TR_HOST_ARM))
+      ltype = getFieldType((J9ROMConstantPoolItem *)romLiterals(), cpIndex);
+#endif
+      }
+
+   setAttributeResult(true, theFieldIsFromLocalClass, ltype, volatileFlag, finalFlag, privateFlag, type, volatileP, isFinal, isPrivate, address);
+
+   return theFieldIsFromLocalClass;
+   }
+
+void
+TR_ResolvedRelocatableJ9JITaaSServerMethod::setAttributeResult(bool isStaticField, bool result, UDATA ltype, I_32 myVolatile, I_32 myFinal, I_32 myPrivate, TR::DataType * type, bool * volatileP, bool * isFinal, bool * isPrivate, void ** fieldOffset)
```

Comment:
This method looks entirely identical to TR_ResolvedRelocatableJ9Method version. Isn't better just to redirect there?

Reply:
setAttributeResult is a protected method of 
TR_ResolvedRelocatableJ9Method. 
TR_ResolvedRelocatableJ9JITaasServerMethod is derived from TR_ResolvedJ9JITaasServerMethod. Hence the class TR_ResolvedRelocatableJ9Method does not come in the hierarchy tree.

Reply:
Reading the code again, there is nothing related to the state of  `TR_ResolvedRelocatableJ9Method` so we could make this method static and public. The same goes for the other two static methods `decodeType` and `getFieldType`